### ISSUE TITLE
Bug in JointLimitRange::validate()

### ISSUE THREAD

**Issue by AlexanderFabisch** [rock-core/base-types#108]
The documentation of this function says "Checks that the provided JointState is within limits for the values that are valid\n@throw OutOfBoundsException if limits are exceeded."

The actual behavior is: the OutOfBoundsException is thrown if the limits are not exceeded, no exception is thrown if the limits are exceeded:

```cpp
void JointLimitRange::validate(const JointState& state) const
{
    std::pair<bool, OutOfBounds> check = verifyValidity(state);
    if (check.first)
        throw check.second;
}
```

`JointLimitRange::verifyValidity()` returns a `std::pair<bool, OutOfBounds>` where pair.first indicates whether the joint state is valid.

**Comment by jmachowinski** [comment]
Yup, that's a bug. As no one noticed it for years, this function looks like dead code...

**Comment by jmachowinski** [comment]
Fixed in 69b7e9d03c0d61b167d0c90d9fdd091eb010bed7

**Comment by AlexanderFabisch** [comment]
Thanks!
